gridcoin-community/Gridcoin-Research
# Upgrade utilities (Update checker and snapshot downloader/application)

These options not available for TestNet but a TestNet snapshot option can be added in future with ease!

Update Checker:

* Check github for latest version tag
* If daemon it will display on screen the update information
* If Qt a Message Box will appear with the information about the release
* Check on start up for new version
* Check every 24 hours via CScheduler

Snapshot Downloader:
Daemon:
* Start gridcoin daemon with argument `-snapshotdownload` and the snapshot process will start and once complete the daemon will continue to start up without any user actions needed.
* Start gridcoin qt with argument `-snapshotdownload` and the snapshot process will start (with Qt Progress updates) and once complete the Qt will continue to start up without any user actions needed
* Request snapshot download via Qt Menu which when confirm the user wishes to do so it will shutdown the wallet and perform the snapshot process and then once complete the user just has to restart the wallet and continue syncing up.

Snapshot Process:
Stage 1) Download snapshot.zip from gridcoin.us server (with realtime updates via % complete and download speed)
Stage 2) Verify the SHA256SUM of snapshot.zip against the SHA256SUM stored on gridcoin.us server
Stage 3) Remove previous blockchain data from the wallet data directory.
Stage 4) Extract the snapshot.zip file to data directory.

It's rather straight forward and works as intended. I've added `_()` to even the Qt side because we wont have the translations loaded for Qt but the daemon side does not use them directly.  I've also adopted the `.lock` file check and `fsbridge` support to this as well.

This is a big PR and look forward to review and requested changes. I'm still looking to refactor and refine some things. This also means no more manual applying of snapshot for users.

Some Pictures:
![Screenshot from 2019-10-25 14-38-25](https://user-images.githubusercontent.com/13292864/67611275-54dfb400-f74d-11e9-9997-90ba3fed438f.png)
![Screenshot from 2019-10-25 14-11-02](https://user-images.githubusercontent.com/13292864/67611282-65902a00-f74d-11e9-998d-d73bfe0882be.png)
![Screenshot from 2019-10-25 14-25-58](https://user-images.githubusercontent.com/13292864/67611295-793b9080-f74d-11e9-8340-1d64b1e06e17.png)
![Screenshot from 2019-10-25 14-34-19](https://user-images.githubusercontent.com/13292864/67611300-86f11600-f74d-11e9-8281-fe57e9e38a95.png)
![Screenshot from 2019-10-25 14-36-12](https://user-images.githubusercontent.com/13292864/67611306-8f495100-f74d-11e9-917b-c4960e2b534d.png)
